torchaudio/functional.py

```diff
@@ -86,16 +89,21 @@ def spectrogram(
 
     # default values are consistent with librosa.core.spectrum._spectrogram
     spec_f = torch.stft(
-        waveform, n_fft, hop_length, win_length, window, True, "reflect", False, True
+        waveform, n_fft=n_fft,
+        hop_length=hop_length, win_length=win_length,
+        window=window, center=True, pad_mode="reflect", normalized=False, onesided=True,
+        return_complex=True,
     )
 
     # unpack batch
-    spec_f = spec_f.reshape(shape[:-1] + spec_f.shape[-3:])
+    spec_f = spec_f.reshape(shape[:-1] + spec_f.shape[-2:])
 
     if normalized:
         spec_f /= window.pow(2.).sum().sqrt()
     if power is not None:
-        spec_f = complex_norm(spec_f, power=power)
+        spec_f = spec_f.abs().pow(power)
+    elif not return_complex:
```

Comment:
shouldn't it be a separate if statement?

Reply:
As I mentioned in the previous reply, when `power` is provided, this function returns the power of spectrogram, which is real value. So `return_complex` only applies when `power` is not applied (the function is returning complex-valued tensor).